harness/determined/exec/prep_container.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import base64
+import io
+import os
+import tarfile
+
+import determined as det
+from determined import constants
+from determined.common.api import certs, request
+
+
+def trial_prep(info: det.ClusterInfo, cert: certs.Cert) -> None:
+    trial_info = det.TrialInfo._from_env()
+    trial_info._to_file()
+
+    resp = request.get(info.master_url, f"api/v1/experiments/{trial_info.experiment_id}/model_def")
+    resp.raise_for_status()
+
+    tgz = base64.b64decode(resp.json()["b64Tgz"])
+
+    with tarfile.open(fileobj=io.BytesIO(tgz), mode="r:gz") as model_def:
+        # Ensure all members of the tarball resolve to subdirectories.
```

Comment:
this is a security concern; you don't want somebody giving you a tarball with a path that's going to try to overwrite `../../../../../../../../etc/sudoers` or something like that.

As I recall, this was a check that the `tarfile` helper function normally applied, so I added it here since I couldn't use the helper function for some reason or another.  Right now this part of the code just moved from `fetch_context.py`.